### DISCUSSION TITLE
Introducing typing for actions

### DISCUSSION THREAD

**Discussion by krzema12** [community/community#32054] [Actions]
I'm the author of https://github.com/typesafegithub/github-actions-typing. The goal of the project is to remove stringly-typed APIs of GitHub Actions, and bring more type-safety. It's realized by adding an `action-types.yml` + a validation workflow, thanks to which automation like e.g. this library https://github.com/typesafegithub/github-workflows-kt can automatically generate type-safe wrappers for using the actions when defining GitHub Action workflows in Kotlin. I can imagine other utilities to make good use of this extra metadata, plus it brings a standardized way of describing types of inputs, replacing what's usually written in READMEs.

I'd like to encourage you to give it a try and share your feedback!

**Comment by Akash1134** [comment]
Hi @krzema12 Thanks for sharing your project. I've added relevant tag to this discussion so that the product team can take a look at it.

**Reply by krzema12** [reply]
I just want to quickly check if there's been any progress on the product team side @Akash1134.

**Reply by krzema12** [reply]
@Akash1134 friendly ping, could I get this idea triaged by the product team please?

**Comment by krzema12** [comment]
CC @ncalteen as we discussed this idea.
